php_phongo.c

```diff
@@ -1591,6 +1607,178 @@ void php_phongo_populate_default_ssl_ctx(php_stream_context *ctx, zval *driverOp
 #undef SET_STRING_CTX
 } /* }}} */
 
+bool php_phongo_apply_rp_options_to_client(mongoc_client_t *client, bson_t *options TSRMLS_DC) /* {{{ */
+{
+	bson_iter_t iter;
+	mongoc_read_prefs_t *new_rp;
+	const mongoc_read_prefs_t *old_rp;
+
+	if (!(old_rp = mongoc_client_get_read_prefs(client))) {
+		MONGOC_WARNING("Client does not have a read preference.");
```

Comment:
The mongoc use of MONGOC_WARNING() for error messaging is problematic.
All we can do is log it and fail. And the failure contains no information as to why it failed. It just failed.

For example, monogc will issue MONGOC_WARNING() as to why a mongoc_bulk_operation_replace_one() was not added to the bulk. The function declaration is void, so its really difficult for a user to determine if an bulk operation has been added to the bulk or not, and doing so programmatically is a nightmare.

From phongo perspective, these log messages are just that.. logs. That are sent to /dev/null unless you turn on 'mongodb.debug' php.ini

I think, when something fails, or we don't do something, or interrupt early, or whatever, we should throw. throw an exception with why that happened.
If it isn't an exception worthy situation, then we should make it a debug or info level log message.
Why? Simply because WARNING runs the risk of the next person after you seeing this code assuming it means something that it doesn't. It doesn't trigger exception. It doesn't abort. It doesn't indicate any sorts of failure, and it doesn't tell the user anything by default.